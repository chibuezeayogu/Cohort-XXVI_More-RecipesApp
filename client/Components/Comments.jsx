import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
