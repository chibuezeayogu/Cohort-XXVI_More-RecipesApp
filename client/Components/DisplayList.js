import React, { Component } from 'react';
import { Link } from 'react-router';

const Displaylist = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <td>
                list ingradients
            </td>
        );
    }
});

export default Displaylist;