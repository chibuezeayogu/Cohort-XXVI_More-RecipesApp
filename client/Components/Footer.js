import React, { Component }from 'react';
import PropTypes from 'prop-types';
import { Footer } from 'react-materialize';


class BottomFooter extends  Component {
    render() {
        return (
            <div>
                <Footer copyrights="&copy 2017 Copyright Text"
                moreLinks={
                    <a className="grey-text text-lighten-4 right" href="#!">Andela Project by Ayogu Chibueze></a>
                }
                
                className='black'
            >
                    
                </Footer>;
            </div>
        );
    }
}

BottomFooter.propTypes = {
    logout: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
    className: PropTypes.string,
    copyrights: PropTypes.string,
};

export default BottomFooter;