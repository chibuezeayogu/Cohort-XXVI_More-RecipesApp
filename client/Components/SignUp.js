import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Textarea, Button, Col} from 'react-materialize';
//require('dotenv').config();
const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;

class SignUp extends  Component{
    constructor(props) {
        super(props);
        this.state = {
            file: '',
            imagePreviewUrl: '',
            profileImageUrl: ''
        };
      }
   
      _handleImageChange(e) {
        e.preventDefault();
    
        let reader = new FileReader();
        let file = e.target.files[0];
        const formData = new FormData;
        formData.append('file', file);
        formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);

        $.ajax({
            type: "POST",
            url: CLOUDINARY_URL,
            dataType: 'json',
            data: formData,
            contentType:false,
            processData: false,
            success: ((data) => {
                this.setState({profileImageUrl:data.secure_url});
            })
        });

        reader.onloadend = () => {
          this.setState({
            file: file,
            imagePreviewUrl: reader.result
          });
        }
    
        reader.readAsDataURL(file)
      }

    handleOnsubmit(e) {
        e.preventDefault();
        const firstname = this.firstname.state.value;
        const lastname = this.lastname.state.value;
        const email = this.email.state.value;
        const password = this.password.state.value;
        const profileImageUrl = this.state.profileImageUrl;
        this.props.createAccount(firstname, lastname, email, password, profileImageUrl);
        this.refs.SignupForm.reset();
        this.setState({
            imagePreviewUrl: ' '
            });
    }

    render() {
        return (
            <div className="container">\
                <div className="col l6 m8 s12 offset-l6 offset-m4">
                <h4 className="center">Create Account</h4>
                <hr/>
                <form className="col l6 m8 s12 offset-l3 offset-m2" onSubmit={(e) => this.handleOnsubmit(e)} ref="SignupForm">
                    <Row>
                        <Input s={6} label="First Name" ref={(input) => {this.firstname=input}}  type='text'><Icon>account_circle</Icon></Input>
                        <Input s={6} label="Last Name" ref={(input) => {this.lastname=input}} type='text'><Icon>account_circle</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={6} label="Email" ref={(input) => {this.email=input}} classtName='validate' type='email'><Icon>mail_outline</Icon></Input>
                        <Input s={6} label="Password" ref={(input) => {this.password=input}} type='password' ><Icon>vpn_key</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={6} id="recipe"  name="file" type='file' ref={(input) => {this.file = input}} onChange={(e)=>this._handleImageChange(e)}><Icon>insert_photo</Icon></Input>
                        <Col s={6}><img className="responsive-img img-wh" src={this.state.imagePreviewUrl?this.state.imagePreviewUrl:""}/></Col>
                    </Row>
                    <hr />
                    <Row>
                        <Button waves='light' className='responsive-img circle right' alt="Profile preview">Sign Up<Icon right>send</Icon></Button> 
                    </Row>
                </form>  
                </div>
            </div>    
        );
    }
}

SignUp.propTypes = {
    handleOnSubmit: PropTypes.func.isRequired,
    _addIngredients:  PropTypes.func.isRequired,
    _addProcedures:  PropTypes.func.isRequired,
    _handleImageChange:  PropTypes.func.isRequired,
    login: PropTypes.func.isRequired,
    //className:
    };

export default SignUp;