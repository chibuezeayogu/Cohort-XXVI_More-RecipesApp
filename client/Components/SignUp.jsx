require('dotenv').config();

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { createAccount } from '../action/actionCreators';
import { validateSignUpForm } from '../util/validateInputs';
import Menu from './Header/Menu.jsx';
import Footer from './Footer/Footer.jsx';
import SmallPreloader from './SmallPreloader.jsx';
import imageToFormData from '../util/ImageUpload';

/**
 * SignUp component for user to create an account
 *
 * @class
 *
 * @extends Component
 *
 */
export class SignUp extends Component {
  /**
   * @description initialize state and binds functiom
   *
   * @constructor
   *
   * @memberOf SignUp
   *
   * @returns {undefined}
   *
   */
  constructor() {
    super();
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      imageUrl: '',
      image: {},
      errors: {},
      disabled: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleOnsubmit = this.handleOnsubmit.bind(this);
    this.handleImageChange = this.handleImageChange.bind(this);
  }

  /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @returns {undefined}
   *
   */
  componentWillMount() {
    if (this.props.userData.isAuthenticated) {
      this.props.history.push('/recipes');
    }
  }

  /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @param {Object} nextProps - nextProps object
   *
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.userData.isAuthenticated) {
      this.props.history.push('/recipes');
    } else {
      this.setState({ disabled: false });
    }
  }

  /**
   * @description Set the state of user input
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @param {Event} event - event object
   *
   * @returns {undefined}
   */
  handleChange(event) {
    event.preventDefault();
    this.setState({ errors: {} });
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description validates image input and sets image state
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @param {Event} event - event objectt
   *
   * @returns {undefined}
   *
   */
  handleImageChange(event) {
    event.preventDefault();
    const file = event.target.files[0];
    this.setState({ errors: {}, image: {} });
    this.setState({ image: file });
  }

  /**
   * @description submits form
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @param {Event} event - event object
   *
   * @returns {undefined}
   *
   */
  handleOnsubmit(event) {
    event.preventDefault();
    const err = validateSignUpForm(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }

    this.setState({ disabled: true });
    const uploadData = imageToFormData(this.state.image);
    delete axios.defaults.headers.common.Authorization;

    axios(uploadData)
      .then((data) => {
        this.setState({ imageUrl: data.data.secure_url });
        this.props.createAccount(this.state);
      }).catch(() => {
        this.setState({ disabled: false });
        Materialize.toast('Error! Please try again', 4000, 'red');
      });
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @returns {undefined} - renders JSX element
   *
   */
  render() {
    return (
      <div className="body">
        <Menu />
        <div className="main">
          <div className="signup-form z-depth-1">
            <div className="col l6 m8 s12 offset-l6 offset-m4">
              <h4 className="center">Create Account</h4>
              <hr />
              <form
                className="col l6 m8 s12 offset-l3 offset-m2"
                onSubmit={this.handleOnsubmit}
              >
                <div className="row">
                  <div className="input-field col s12">
                    <i className="fa fa-user-circle prefix" aria-hidden="true" />
                    <input
                      id="firstName"
                      name="firstName"
                      type="text"
                      value={this.state.firstName}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="firstName" className="active">
                      First Name
                    </label>
                  </div>
                  <span className="right red-text error-font error-margin">
                    {this.state.errors.firstNameError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <i className="fa fa-user-circle prefix" aria-hidden="true" />
                    <input
                      id="lastName"
                      name="lastName"
                      type="text"
                      value={this.state.lastName}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="lastName" className="active">
                      Last Name
                    </label>
                  </div>
                  <span className="right red-text error-font error-margin">
                    {this.state.errors.lastNameError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <i className="fa fa-envelope prefix" aria-hidden="true" />
                    <input
                      id="email"
                      name="email"
                      type="text"
                      value={this.state.email}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="email" className="active">Email</label>
                  </div>
                  <span className="right red-text error-font error-margin">
                    {this.state.errors.emailError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <i className="fa fa-key prefix" aria-hidden="true" />
                    <input
                      id="password"
                      name="password"
                      type="password"
                      value={this.state.password}
                      onChange={this.handleChange}
                      autoComplete="off"
                    />
                    <label htmlFor="password">Password</label>
                  </div>
                  <span className="right red-text error-font error-margin">
                    {this.state.errors.passwordError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <i className="fa fa-picture-o prefix" aria-hidden="true" />
                    <input
                      id="image"
                      type="file"
                      name="profileImage"
                      onChange={this.handleImageChange}
                    />
                  </div>
                  <span className="right red-text error-font error-margin">
                    {this.state.errors.imageError}
                  </span>
                </div>
                <div className="row right">
                  <button
                    className="btn right green"
                    type="submit"
                    name="action"
                  > Sign Up
                  </button>
                </div>
                <div className="row center">
                  <div>
                    {this.state.disabled ?
                      <SmallPreloader />
                    :
                    ''
                    }
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

SignUp.propTypes = {
  userData: PropTypes.shape({
    isAuthenticated: PropTypes.bool.isRequired
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  createAccount: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  userData: state.userData
});

export default connect(mapStateToProps, { createAccount })(SignUp);
