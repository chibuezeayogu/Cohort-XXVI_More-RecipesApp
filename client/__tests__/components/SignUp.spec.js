import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { SignUp } from '../../Components/SignUp';
import axios from 'axios';
import imageToFormData from '../../util/ImageUpload';


const props = {
  userData: {
    isAuthenticated: true,
    currentUser: { id: 1, email: 'chibuezeayogu@hotmail.com' }
  },
  history: {
    push: jest.fn()
  },
  createAccount: jest.fn()
};

const event = {
  preventDefault: jest.fn(),
  target: {
    name: 'email',
    value: 'chibuezeayogu@hotmail.com',
    files: [{}]
  }
};

const state = {
  firstName: 'Chibueze',
  lastName: 'Ayogu',
  email: 'chibuezeayogu@hotmail.com',
  password: 'Abc12345.@',
  imageUrl: 'https://res.cloudinary.com/chibuezeayogu/image/upload/' +
    'v1517769827/tvad0agtwxzucqyokktf.jpg',
  errors: {},
  disabled: false
};

describe('<SignUp />', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<SignUp {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<SignUp />', () => {
  it('should call handleChange function', () => {
    const wrapper = shallow(<SignUp {...props} {...state} />);
    wrapper.find('#email').simulate('change', event);
    expect(wrapper.instance().state.email).toBe('chibuezeayogu@hotmail.com');
  }
  );
});

describe('<SignUp />', () => {
  it('should call handleImageChange function', () => {
    const wrapper = shallow(<SignUp {...props} {...state} />);
    wrapper.find('#image').simulate('change', event);
    expect(wrapper.instance().state.image).toEqual({});
  });
});

describe('componentWillReceiveProps', () => {
  it('should update component with new props', () => {
    const wrapper = shallow(<SignUp {...props} />);
    const componentWillReceivePropsSpy = jest
      .spyOn(wrapper.instance(), 'componentWillReceiveProps');
    wrapper.setProps({
      userData: {
        isAuthenticated: true,
        currentUser: { id: 1, email: 'chibuezeayogu@hotmail.com' }
      }
    });
    wrapper.setState({ disabled: false });
    expect(componentWillReceivePropsSpy).toHaveBeenCalled();
  });
});

describe('HandleOnSubmit function', () => {
  it('should return error if input fields are empty', () => {
    const wrapper = shallow(<SignUp {...props} {...state} />);
    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().state.errors.firstNameError)
      .toBe('first name is required');
    expect(wrapper.instance().state.errors.lastNameError)
      .toBe('last name is required');
    expect(wrapper.instance().state.errors.emailError)
      .toBe('email is required');
    expect(wrapper.instance().state.errors.passwordError)
      .toBe('password is required');
    expect(wrapper.instance().state.errors.imageError)
      .toBe('select a valid file type');
  });

  it('should return error for input did not meet the required lenght', () => {
    const wrapper = shallow(<SignUp {...props} {...state} />);
    wrapper.setState({
      firstName: 'ch',
      lastName: 'Ne',
      email: 'la',
      password: 'Noooo',
    });
    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().state.errors.firstNameError)
      .toBe('first name must be at least 3 characters long');
    expect(wrapper.instance().state.errors.lastNameError)
      .toBe('last name must be at least 3 characters long');
    expect(wrapper.instance().state.errors.emailError)
      .toBe('email is not valide');
    expect(wrapper.instance().state.errors.passwordError)
      .toBe('password must contain `uppercase, lowercase, number, special character`');
  });

  it('should sumbit form if all conditions are meet', () => {
    const wrapper = shallow(<SignUp {...props} {...state} />);

    wrapper.setState({
      firstName: 'chibueze',
      lastName: 'Nelson',
      email: 'chibueze@hotmail.com',
      password: 'Absc.123.@',
      image: { type: 'image/png' },
    });
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(wrapper.instance().state.disabled).toEqual(true);
  });
});

