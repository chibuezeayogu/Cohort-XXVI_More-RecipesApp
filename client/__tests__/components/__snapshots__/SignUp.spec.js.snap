// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body"
  >
    <withRouter(Menu) />
    <div
      className="main"
    >
      <div
        className="signup-form z-depth-1"
      >
        <div
          className="col l6 m8 s12 offset-l6 offset-m4"
        >
          <h4
            className="center"
          >
            Create Account
          </h4>
          <hr />
          <form
            className="col l6 m8 s12 offset-l3 offset-m2"
            onSubmit={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user-circle prefix"
                />
                <input
                  id="firstName"
                  name="firstName"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label
                  className="active"
                  htmlFor="firstName"
                >
                  First Name
                </label>
              </div>
              <span
                className="right red-text error-font error-margin"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user-circle prefix"
                />
                <input
                  id="lastName"
                  name="lastName"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label
                  className="active"
                  htmlFor="lastName"
                >
                  Last Name
                </label>
              </div>
              <span
                className="right red-text error-font error-margin"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-envelope prefix"
                />
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label
                  className="active"
                  htmlFor="email"
                >
                  Email
                </label>
              </div>
              <span
                className="right red-text error-font error-margin"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-key prefix"
                />
                <input
                  autoComplete="off"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
                <label
                  htmlFor="password"
                >
                  Password
                </label>
              </div>
              <span
                className="right red-text error-font error-margin"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-picture-o prefix"
                />
                <input
                  id="image"
                  name="profileImage"
                  onChange={[Function]}
                  type="file"
                />
              </div>
              <span
                className="right red-text error-font error-margin"
              />
            </div>
            <div
              className="row right"
            >
              <button
                className="btn right green"
                name="action"
                type="submit"
              >
                 Sign Up
              </button>
            </div>
            <div
              className="row center"
            >
              <div>
                
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body"
    >
      <withRouter(Menu) />
      <div
        className="main"
      >
        <div
          className="signup-form z-depth-1"
        >
          <div
            className="col l6 m8 s12 offset-l6 offset-m4"
          >
            <h4
              className="center"
            >
              Create Account
            </h4>
            <hr />
            <form
              className="col l6 m8 s12 offset-l3 offset-m2"
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user-circle prefix"
                  />
                  <input
                    id="firstName"
                    name="firstName"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <label
                    className="active"
                    htmlFor="firstName"
                  >
                    First Name
                  </label>
                </div>
                <span
                  className="right red-text error-font error-margin"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user-circle prefix"
                  />
                  <input
                    id="lastName"
                    name="lastName"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <label
                    className="active"
                    htmlFor="lastName"
                  >
                    Last Name
                  </label>
                </div>
                <span
                  className="right red-text error-font error-margin"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-envelope prefix"
                  />
                  <input
                    id="email"
                    name="email"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <label
                    className="active"
                    htmlFor="email"
                  >
                    Email
                  </label>
                </div>
                <span
                  className="right red-text error-font error-margin"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-key prefix"
                  />
                  <input
                    autoComplete="off"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                  <label
                    htmlFor="password"
                  >
                    Password
                  </label>
                </div>
                <span
                  className="right red-text error-font error-margin"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-picture-o prefix"
                  />
                  <input
                    id="image"
                    name="profileImage"
                    onChange={[Function]}
                    type="file"
                  />
                </div>
                <span
                  className="right red-text error-font error-margin"
                />
              </div>
              <div
                className="row right"
              >
                <button
                  className="btn right green"
                  name="action"
                  type="submit"
                >
                   Sign Up
                </button>
              </div>
              <div
                className="row center"
              >
                <div>
                  
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SignUp
        createAccount={[MockFunction]}
        history={
          Object {
            "push": [MockFunction] {
              "calls": Array [
                Array [
                  "/recipes",
                ],
              ],
            },
          }
        }
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SignUp {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "handleImageChange": [Function],
        "handleOnsubmit": [Function],
        "props": Object {
          "createAccount": [MockFunction],
          "history": Object {
            "push": [MockFunction] {
              "calls": Array [
                Array [
                  "/recipes",
                ],
              ],
            },
          },
          "userData": Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          },
        },
        "refs": Object {},
        "state": Object {
          "disabled": false,
          "email": "",
          "errors": Object {},
          "firstName": "",
          "image": Object {},
          "imageUrl": "",
          "lastName": "",
          "password": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body"
        >
          <withRouter(Menu) />
          <div
            className="main"
          >
            <div
              className="signup-form z-depth-1"
            >
              <div
                className="col l6 m8 s12 offset-l6 offset-m4"
              >
                <h4
                  className="center"
                >
                  Create Account
                </h4>
                <hr />
                <form
                  className="col l6 m8 s12 offset-l3 offset-m2"
                  onSubmit={[Function]}
                >
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user-circle prefix"
                      />
                      <input
                        id="firstName"
                        name="firstName"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="firstName"
                      >
                        First Name
                      </label>
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user-circle prefix"
                      />
                      <input
                        id="lastName"
                        name="lastName"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="lastName"
                      >
                        Last Name
                      </label>
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-envelope prefix"
                      />
                      <input
                        id="email"
                        name="email"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="email"
                      >
                        Email
                      </label>
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-key prefix"
                      />
                      <input
                        autoComplete="off"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        value=""
                      />
                      <label
                        htmlFor="password"
                      >
                        Password
                      </label>
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-picture-o prefix"
                      />
                      <input
                        id="image"
                        name="profileImage"
                        onChange={[Function]}
                        type="file"
                      />
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row right"
                  >
                    <button
                      className="btn right green"
                      name="action"
                      type="submit"
                    >
                       Sign Up
                    </button>
                  </div>
                  <div
                    className="row center"
                  >
                    <div>
                      
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body"
        >
          <withRouter(Menu) />
          <div
            className="main"
          >
            <div
              className="signup-form z-depth-1"
            >
              <div
                className="col l6 m8 s12 offset-l6 offset-m4"
              >
                <h4
                  className="center"
                >
                  Create Account
                </h4>
                <hr />
                <form
                  className="col l6 m8 s12 offset-l3 offset-m2"
                  onSubmit={[Function]}
                >
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user-circle prefix"
                      />
                      <input
                        id="firstName"
                        name="firstName"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="firstName"
                      >
                        First Name
                      </label>
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user-circle prefix"
                      />
                      <input
                        id="lastName"
                        name="lastName"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="lastName"
                      >
                        Last Name
                      </label>
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-envelope prefix"
                      />
                      <input
                        id="email"
                        name="email"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="email"
                      >
                        Email
                      </label>
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-key prefix"
                      />
                      <input
                        autoComplete="off"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        value=""
                      />
                      <label
                        htmlFor="password"
                      >
                        Password
                      </label>
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-picture-o prefix"
                      />
                      <input
                        id="image"
                        name="profileImage"
                        onChange={[Function]}
                        type="file"
                      />
                    </div>
                    <span
                      className="right red-text error-font error-margin"
                    />
                  </div>
                  <div
                    className="row right"
                  >
                    <button
                      className="btn right green"
                      name="action"
                      type="submit"
                    >
                       Sign Up
                    </button>
                  </div>
                  <div
                    className="row center"
                  >
                    <div>
                      
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignUp
    createAccount={[MockFunction]}
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/recipes",
            ],
          ],
        },
      }
    }
    userData={
      Object {
        "currentUser": Object {
          "email": "chibuezeayogu@hotmail.com",
          "id": 1,
        },
        "isAuthenticated": true,
      }
    }
  />,
}
`;
